README


This archive contains the distribution WebAPI-DBIC,
version 0.003001:

  A composable RESTful JSON API to DBIx::Class schemas using roles and Web::Machine

This software is copyright (c) 2014 by Tim Bunce.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.022.



Changes
Changes file for WebAPI::DBIC

0.003001  2014-12-24 15:34:48+00:00 Europe/Dublin (TRIAL RELEASE)

 - Major update to WebAPI::DBIC::WebApp. Added RouteMaker and TypeNamer (experimental).
 - Added type_name_inflect to control inflection of paths and type names
 - Auth handles more kinds of schema connection_info data
 - Improved compatibility for JSON API (still a work in progress)
 - Fixed fixture filenames to be windows compatable

0.002007  2014-12-14 22:29:58+00:00 Europe/Dublin

 - Add dependency on Cpanel::JSON::XS to preserve numerics in JSON #21
 - Add a CONTRIBUTING file, based on ETHER's

0.002006  2014-12-14 16:41:01+00:00 Europe/Dublin

  - Added WebAPI::DBIC::Route class and refactored ::WebApp to use it.
  - Workaround Test::Differences context size bug RT#100912
  - Switch expected test results to use non-stringified integers

0.002005  2014-12-11 21:23:46+00:00 Europe/Dublin

  - For ResultSets with no primary key defined warn and skip item routes.
  - Unbundle HAL browser into Alien::Web::HalBrowser (Vanstyn)
  - Make Test::Compile version insufficiency more clear

0.002004  2014-12-06 18:49:30+00:00 Europe/Dublin

  - Renamed WebAPI::DBIC::Resource::Role::Foo$TYPE -> ...::Resource::$TYPE::Role::Foo
  - Renamed test files similarly to split into per-media-type directories
  - Fix handling of JSON boolean in WebAPI::DBIC::Resource::HAL::Role::Root

0.002003  2014-12-06 12:59:36+00:00 Europe/Dublin

  - Possibly incompatible change: paths are now based on the schema source
    (eg class) name, rather than the result source (eg table) name.
  - Moved handling of root (/) to a Web::Machine resource class.
  - Further generalization of WebAPI::DBIC::WebApp.
  - Deprecate order=foo%20desc in favor of sort=-foo, and tighten spec.
  - Start support for foo[bar]=... style of parameter naming.
  - Migrate tests to DBIx::Class::Fixtures to enable loading in demo app.

0.002002  2014-11-25 15:25:29+00:00 Europe/Dublin

  - Fixed webapi-dbic-demo.psgi (Mike Francis)
  - Fixed use of DSN as Auth Realm (Mike Francis)
  - Added compilation tests for *.psgi files.

0.002001  2014-11-24 19:57:02+00:00 Europe/Dublin

  - Require DBIx::Class 0.08250+ (#14)
  - Added autodie to cpanfile for Perl 5.10
  - Fixed stability of t/30-ordering-hal.t output

0.002000  2014-11-17 23:01:09+00:00 Europe/Dublin

  - Added initial support for JSON API application/vnd.api+json media type
  - Added support for arbitrary depth/complexity of prefetch (Mike Francis)

0.001010  2014-11-12 14:05:00+00:00 Europe/Dublin (TRIAL RELEASE)

  - Prefetch now supports has_many relationships on item and set resources.
  - Major refactoring to separate out HAL support ready for JSON API.
  - Changed (some) tests to use a simpler 'golden result' scheme.

0.001008  2014-11-03 21:05:28+00:00 Europe/Dublin

  - Improved error reporting for prefetch attribute.
  - Ignore empty params.
  - Add "did you mean me.$param" to exception for params containing "."

0.001008  2014-11-03 09:28:33+00:00 Europe/Dublin (TRIAL RELEASE)

  - Added skip_dirty_check to WebAPI::DBIC::Resource::Role::ItemWritable
  - Added a COMPARISONS section to docs (and renamed INTEGRATING to INTEGRATIONS)
  - Added minimum versions to some prerequisites in cpanfile

0.001007  2014-08-18 12:04:57+01:00 Europe/Dublin

  - Add Catalyst integration example (Arthur Axel fREW Schmidt)
  - Switch from Module::Pluggable to Test::Compile for t/00-compile.t
  - Doc updates re router vs app integration, and transparency.

0.001006  2014-08-17 16:04:19+01:00 Europe/Dublin

  - Abstracted the router interface.
  - Renamed some classes.
  - Added more documentation.

0.001005  2014-08-17 00:43:49+01:00 Europe/Dublin (TRIAL RELEASE)

  - Added tests and test schema tables for multi-field keys (mfr++)
  - Much improved handling of tables with multi-field keys /foo/key1/key2/keyN/
  - Improved handling of code-ref based joins
  - cached relationship _link building information

0.001004  2014-07-28 11:42:41+02:00 Europe/Paris

  - Minor documentation and toolchain  fixes.

0.001003  July 28 2014

  - the first release


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.022.
use strict;
use warnings;

use 5.010;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "A composable RESTful JSON API to DBIx::Class schemas using roles and Web::Machine",
  "AUTHOR" => "Tim Bunce <Tim.Bunce\@pobox.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "WebAPI-DBIC",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010",
  "NAME" => "WebAPI::DBIC",
  "PREREQ_PM" => {
    "Alien::Web::HalBrowser" => 0,
    "DBD::SQLite" => "1.46",
    "DBIx::Class" => "0.08250",
    "Data::Dumper::Concise" => 0,
    "HTTP::Headers::ActionPack" => 0,
    "HTTP::Throwable" => 0,
    "JSON::MaybeXS" => 0,
    "Lingua::EN::Inflect" => "1.894",
    "Lingua::EN::Inflect::Number" => "1.11",
    "List::Util" => 0,
    "Module::Runtime" => 0,
    "Moo" => "1.001000",
    "Path::Router" => "0.13",
    "Plack" => "1.0033",
    "Plack::App::File" => 0,
    "Plack::App::Path::Router" => "0.06",
    "SQL::Translator" => "0.11018",
    "Scalar::Util" => 0,
    "String::CamelCase" => 0,
    "Sub::Exporter" => 0,
    "Sub::Quote" => 0,
    "Sub::Util" => 0,
    "Try::Tiny" => 0,
    "URI" => 0,
    "Web::Machine" => "0.15",
    "namespace::clean" => 0,
    "parent" => 0
  },
  "TEST_REQUIRES" => {
    "Cpanel::JSON::XS" => ">= 3.0110, != 3.0112",
    "DBIx::Class::Fixtures" => "1.001025",
    "Data::Printer" => 0,
    "Module::Pluggable" => 0,
    "Sort::Key" => 0,
    "Test::Compile" => "1.1.0",
    "Test::DBIx::Class" => "0.43",
    "Test::HTTP::Response" => 0,
    "Test::More" => "0.98",
    "Test::Most" => 0,
    "Test::Pod" => 0,
    "ToolSet" => 0,
    "autodie" => 0
  },
  "VERSION" => "0.003001",
  "test" => {
    "TESTS" => "t/*.t t/media-hal/*.t t/media-jsonapi/*.t"
  }
);


my %FallbackPrereqs = (
  "Alien::Web::HalBrowser" => 0,
  "Cpanel::JSON::XS" => ">= 3.0110, != 3.0112",
  "DBD::SQLite" => "1.46",
  "DBIx::Class" => "0.08250",
  "DBIx::Class::Fixtures" => "1.001025",
  "Data::Dumper::Concise" => 0,
  "Data::Printer" => 0,
  "ExtUtils::MakeMaker" => 0,
  "HTTP::Headers::ActionPack" => 0,
  "HTTP::Throwable" => 0,
  "JSON::MaybeXS" => 0,
  "Lingua::EN::Inflect" => "1.894",
  "Lingua::EN::Inflect::Number" => "1.11",
  "List::Util" => 0,
  "Module::Pluggable" => 0,
  "Module::Runtime" => 0,
  "Moo" => "1.001000",
  "Path::Router" => "0.13",
  "Plack" => "1.0033",
  "Plack::App::File" => 0,
  "Plack::App::Path::Router" => "0.06",
  "SQL::Translator" => "0.11018",
  "Scalar::Util" => 0,
  "Sort::Key" => 0,
  "String::CamelCase" => 0,
  "Sub::Exporter" => 0,
  "Sub::Quote" => 0,
  "Sub::Util" => 0,
  "Test::Compile" => "1.1.0",
  "Test::DBIx::Class" => "0.43",
  "Test::HTTP::Response" => 0,
  "Test::More" => "0.98",
  "Test::Most" => 0,
  "Test::Pod" => 0,
  "ToolSet" => 0,
  "Try::Tiny" => 0,
  "URI" => 0,
  "Web::Machine" => "0.15",
  "autodie" => 0,
  "namespace::clean" => 0,
  "parent" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



